segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java

```diff
@@ -92,15 +92,9 @@ private static Map.Entry<Long, Long> applySearch(StreamSegmentStore store, Strin
                 case Cache: // fallthrough
                 case Storage:
                     firstElement.requestContent(TIMEOUT);
-                    AtomicReference<IndexEntry> entry = new AtomicReference<>();
-                    firstElement.getContent().thenAccept(content -> {
-                        entry.set(IndexEntry.fromBytes(content));
-                    });
-                    if (entry.get() != null) {
-                        return entry.get().getOffset();
-                    } else {
-                        throw new IllegalStateException(String.format("Unable to read data from index segment {} of type {}.", segment, firstElement.getType()));
-                    }
+                    BufferView content = firstElement.getContent().join();
```

Comment:
In case if requestContesnt is not able to get result by TIMEOUT.  This Join() will be blocked.
Can we use  something like Futures.join(firstElement.getContent(), 2, TimeUnit.SECONDS);  and handle timeout of this join to throw a retriable exception?

Reply:
As per my understanding, In that case, requestContent will through an exception and flow will not reach to join call.

Reply:
yes and same can be confirmed by running the system tests.